packages/fs-detectors/src/detect-builders.ts

```diff
@@ -305,6 +305,27 @@ export async function detectBuilders(
     };
   }
 
+  // Exclude the middleware builder for Next.js apps since @vercel/next
+  // will build middlewares.
+  //
+  // While maybeGetApiBuilder() excludes the middleware builder, however,
+  // we need to check if it's a Next.js app here again for the case where
+  // `projectSettings.framework == null`.
+  if (
+    framework === null &&
+    detectFrontBuilder(pkg, files, usedFunctions, fallbackEntrypoint, options)
+      .use === '@vercel/next' &&
+    apiBuilders.length > 0
+  ) {
+    apiBuilders = apiBuilders.filter(builder => {
+      const isMiddlewareBuilder =
+        builder.use === '@vercel/node' &&
+        builder.config?.middleware &&
+        (builder.src === 'middleware.js' || builder.src === 'middleware.ts');
```

Comment:
Is there a scenario where `builder.config?.middleware` is true but the `src !== 'middleware.js/ts`? Seems like checking for `builder.config?.middleware` should be sufficient.